### PR TITLE
Feature/hide endpoints

### PR DESCRIPTION
**Pull request by Matt-Guest**
### What

Hide authenticated endpoints

### How to review

Review the changes, run unit tests and acceptance tests (see https://github.com/ONSdigital/dp-api-tests/pull/33/files)

### Who can review

anyone

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package main
```
Comment (CarlHembrough) [inline main.go]: I am guessing this was added by accident. If its something you want to keep, consider adding it to the cmd directory

File: Makefile [hunk 2]
```diff
@@ -13,6 +13,9 @@ build:
 	go build -o $(BUILD_ARCH)/$(BIN_DIR)/$(MAIN) cmd/$(MAIN)/main.go
 debug:
 	HUMAN_LOG=1 go run cmd/$(MAIN)/main.go
+acceptance:
+	SECRET_KEY=0C30662F-6CF6-43B0-A96A-954772267FF5 MONGODB_IMPORTS_DATABASE=test \
```
Comment (CarlHembrough) [inline Makefile]: Can we just add this key as a default in the config?

File: api/authentication.go [hunk 3]
```diff
@@ -22,28 +23,16 @@ func (a *Authenticator) Check(handle func(http.ResponseWriter, *http.Request)) h
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		key := r.Header.Get(a.headerName)
 		if key == "" {
-			http.Error(w, "No authentication header provided", http.StatusForbidden)
-			log.Error(errors.New("client missing token"), log.Data{"header": a.headerName})
+			http.Error(w, genericError, http.StatusNotFound)
```
Comment (CarlHembrough) [inline api/authentication.go]: The name `genericError` was a bit confusing as to its purpose. Consider calling it something like `notFoundErr`

File: api/authentication_test.go [hunk 4]
```diff
@@ -15,7 +16,7 @@ func TestMiddleWareAuthenticationReturnsForbidden(t *testing.T) {
 		So(err, ShouldBeNil)
 		w := httptest.NewRecorder()
 		auth.Check(mockHTTPHandler).ServeHTTP(w, r)
-		So(w.Code, ShouldEqual, http.StatusForbidden)
+		So(w.Code, ShouldEqual, http.StatusNotFound)
```
Comment (CarlHembrough) [inline api/authentication_test.go]: Update the convey message in this test to reflect the change

File: api/authentication_test.go [hunk 5]
```diff
@@ -28,7 +29,7 @@ func TestMiddleWareAuthenticationReturnsUnauthorised(t *testing.T) {
 		So(err, ShouldBeNil)
 		w := httptest.NewRecorder()
 		auth.Check(mockHTTPHandler).ServeHTTP(w, r)
-		So(w.Code, ShouldEqual, http.StatusUnauthorized)
+		So(w.Code, ShouldEqual, http.StatusNotFound)
```
Comment (CarlHembrough) [inline api/authentication_test.go]: Update the convey message in this test to reflect the change

File: api/api.go [hunk 6]
```diff
@@ -35,17 +37,22 @@ type JobService interface {
...
-	auth := NewAuthenticator(secretKey, "internal-token")
+	auth := NewAuthenticator(secretKey, "Internal-Token")
 
 	// External API for florence
-	api.router.Path("/jobs").Methods("POST").HandlerFunc(api.addJob)
-	api.router.Path("/jobs").Methods("GET").HandlerFunc(api.getJobs).Queries()
-	api.router.Path("/jobs/{id}").Methods("GET").HandlerFunc(api.getJob)
-	api.router.Path("/jobs/{id}").Methods("PUT").HandlerFunc(auth.ManualCheck(api.updateJob))
-	api.router.Path("/jobs/{id}/files").Methods("PUT").HandlerFunc(api.addUploadedFile)
+	api.router.Path("/jobs").Methods("POST").HandlerFunc(auth.Check(api.addJob))
```
Comment (CarlHembrough) [inline api/api.go]: consider adding tests to check that each endpoint actually has authentication applied

File: main.go (hunk 1 above)
Comment (Matt-Guest) [reply main.go]: Yes, I removed it.

File: api/api.go (hunk 6 above)
Comment (Matt-Guest) [reply api/api.go]: I added a set of tests to check the authentication is applied.

File: Makefile (hunk 2 above)
Comment (Matt-Guest) [reply Makefile]: done

File: api/authentication.go (hunk 3 above)
Comment (Matt-Guest) [reply api/authentication.go]: done

File: api/authentication_test.go (hunk 4 above)
Comment (Matt-Guest) [reply api/authentication_test.go]: done

File: api/authentication_test.go (hunk 5 above)
Comment (Matt-Guest) [reply api/authentication_test.go]: done

**Pull request merged**